src/correction.cc

```diff
@@ -149,20 +165,33 @@ namespace {
             return bins->n; // the default value is stored at the end of the content array, after the last bin
           case _FlowBehavior::clamp:
             return value < bins->low ? 0 : bins->n - 1; // assuming we always have at least 1 bin
+          case _FlowBehavior::wrap:
+            break;
           case _FlowBehavior::error:
             const std::string belowOrAbove = value < bins->low ? "below" : "above";
             auto msg = "Index " + belowOrAbove + " bounds in " + name + " for input argument " + std::to_string(variableIdx) + " value: " + std::to_string(value);
             throw std::runtime_error(std::move(msg));
         }
       }
 
-      std::size_t binIdx = bins->n * ((value - bins->low) / (bins->high - bins->low));
+      double norm_value = ((value - bins->low) / (bins->high - bins->low));
+      if (flow == _FlowBehavior::wrap) {
+        norm_value -= std::floor(norm_value);
+      }
+      std::size_t binIdx = bins->n * norm_value;
```

Comment:
What if, rather than using `std::floor` to wrap, we use `binIdx = binIdx % bins->n` for `_FlowBehavior::wrap`?

Reply:
Thanks for the suggestion! The major limitation with this is that `norm_value` can also be negative, so the cast from signed `double` to the integer `std::size_t` in `std::size_t binIdx = bins->n * norm_value` would lead to undefined behavior. For positive values is mostly works, but there are some caveats there as well, for example if `bins->n * norm_value` overflows past `ULONG_MAX`.

Another thing to consider is that a calculation based on modulo arithmetic would anyway only apply to uniform binning, whereas the normalize-and-floor approach can also work for nonequidistant bin edges.